pkg/containertools/runner.go

```diff
@@ -22,14 +22,53 @@ type CommandRunner interface {
 type ContainerCommandRunner struct {
 	logger        *logrus.Entry
 	containerTool ContainerTool
+	config        *RunnerConfig
+}
+
+type RunnerConfig struct {
+	SkipTLS bool
+}
+
+type RunnerOption func(config *RunnerConfig)
+
+func SkipTLS(skip bool) RunnerOption {
+	return func(config *RunnerConfig) {
+		config.SkipTLS = skip
+	}
+}
+
+func (r *RunnerConfig) apply(options []RunnerOption) {
+	for _, option := range options {
+		option(r)
+	}
+}
+
+func (r *ContainerCommandRunner) argsForCmd(cmd string, args ...string) []string {
```

Comment:
👍

Reply:
an alternative approach would be to have a `CommandBuilder` abstraction to incrementally build out a command and execute it.